### ISSUE TITLE
Remove dwcCatalogNumber

### ISSUE THREAD

**Issue by cgendreau** [DINA-Web/collection-specs#134]
`dwcCatalogNumber` is replaced by `materialSampleName` since some material sample don't have a catalogNumber.

**Comment by dshorthouse** [comment]
So, you've eliminated `dwcCatalogNumber` but kept `dwcOtherCatalogNumbers`? Would it not make sense then to call the latter, `otherMaterialSampleNames` to better correspond with this change?

**Comment by cgendreau** [comment]
Yes, it should remain consistent but it doesn't imply the same amount of work on the implementation side so it will be done in 2 stages (if we proceed, this is a ticket, nothing is actually eliminated yet).
